packages/core/useIntersectionObserver/index.ts

```diff
@@ -1,12 +1,19 @@
-import { watch } from 'vue-demi'
-import { noop, tryOnScopeDispose } from '@vueuse/shared'
+import { unref, watch } from 'vue-demi'
+import { isBoolean, noop, tryOnScopeDispose } from '@vueuse/shared'
+import type { MaybeRef } from '@vueuse/shared'
 import type { ConfigurableWindow } from '../_configurable'
 import { defaultWindow } from '../_configurable'
 import type { MaybeElementRef } from '../unrefElement'
 import { unrefElement } from '../unrefElement'
 import { useSupported } from '../useSupported'
 
 export interface UseIntersectionObserverOptions extends ConfigurableWindow {
+  /**
+   * Set to `false` if you want to enable only when needed.
+   * @default undefined
+   */
+  enabled?: MaybeRef<boolean>
```

Comment:
I would think it might be better to implement the `Pausiable` interface with a `immediate` option 

https://github.com/mini-ghost/vueuse/blob/3c728cb2630b3afb609da4e350cd5de17686af63/packages/shared/utils/types.ts#L78-L94

Reply:
Perhaps using `Stoppable` would be closer to the current interface?

Reply:
But if we implement the `Pausable` interface with the `immediate` option, developer using directive will not be able to call `pause` and `resume`. If the developer sets immediate: false then he will never be able to resume the IntersectionObserver.

Reply:
Sorry I am not sure if I got what you mean. To have a `Pausable` interface is to make toggling that behavior more explicit. I think it could work the same as `enabled.value`.